Makefile

```diff
@@ -62,6 +62,7 @@ SAIL_DEFAULT_INST += riscv_insts_vext_red.sail
 SAIL_DEFAULT_INST += riscv_insts_vext_fp_red.sail
 SAIL_DEFAULT_INST += riscv_insts_zicbom.sail
 SAIL_DEFAULT_INST += riscv_insts_zicboz.sail
+SAIL_DEFAULT_INST += riscv_insts_zicbop.sail
```

Comment:
This is going to have no effect if you add it here because these instructions overlap `ori` which will already have handled them in `riscv_insts_base.sail` above. This file needs to come before that.

Honestly it feels a bit fragile to just rely on the file order and silently break if you get it wrong though. I don't know if @Alasdair has any good ideas to make it robust.

Reply:
I would like to add a check that file order doesn't matter.

You could take all the decode clauses, and automatically change them so `file1.sail` generates `0b001`, `file2.sail` generates `0b010`, `file3.sail` generates `0b100`, then ask the Sail SMT backend if the sum of each files decode clauses has more than a single bit set. Would likely be a bit heavyweight for a check that would run every single invocation though.

Reply:
I guess the downside is that you'd need to modify the `ORI` decoder so that it excludes these hints somehow.

I guess you could avoid that by explicitly adding an `encdec_hints` or something and checking that first. Kind of rubbish though.

Alternative add some kind of explicit priority for each clause and have the compiler reorder them. Though that kind of thing usually ends up [like CSS z-index](https://blog.frankmtaylor.com/wp-content/uploads/2019/05/z-index.jpg).

Reply:
seems to be a discussion about the positional relationship of scattered clauses. I really can't think of a good solution. At least grammatically, I can only think of adding something like `#[index = 1]` before the clause